### ISSUE TITLE
dynamo/fx doesn't honor 'non-persistent' buffers

### ISSUE THREAD

**Issue by wconstab** [pytorch/pytorch#123411]
need to make a repro for this, but I found that when using torch.export + pippy + fx graph-splitter on a model that has a non-persistent buffer, the buffer was persistent by the time it came out the tracing system.

To be clear,

orig_mod.buf <-- came from .register_buffer("buf", _, persistent=False)

some tracing happened,

then finally traced_mod.buf exists as a buffer, but does not exist in `traced_mod._non_persistent_buffers_set`

The problem appears to originate in dynamo layer not sticking this info onto the root graph,

and again as a bug in the fx graph splitter (fx/graph_module.py:234) which does not care about 'non-persistent-ness' when propagating a buffer from the orig mod to the split mod.

cc @kwen2501 @yf225 @jansel

**Comment by kwen2501** [comment]
Thanks for digging the problem!

When I run the mini repro, even the first assertion fails:
```
    assert 'freqs_cis' in cm._buffers
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

**Comment by wconstab** [comment]
Yea, I think both are good to fix.

Note, I won't be blocked by this, I will have to make freqs_cis persistent anyway for meta inkt from seed ckpt to work

**Comment by wz337** [comment]
I can repro the same issue without PiPPy or any parallelism. 

```
class ModelWithBuffers(nn.Module):
    def __init__(self):
        super().__init__()
        self.lin1 = nn.Linear(4, 4)
        self.lin2 = nn.Linear(4, 4)
        self.register_buffer(
            "non_persistent_buffer", torch.randn((4,)), persistent=False
        )
        self.register_buffer("persistent_buffer", torch.randn((4,)), persistent=True)
        self.weight = nn.Parameter(torch.randn((4, 4)))

model = ModelWithBuffers()
model = torch.compile(model)
optim = torch.optim.Adam(model.parameters())
# prints without compiled: model._non_persistent_buffers_set={'non_persistent_buffer'}
# prints with compiled: model._non_persistent_buffers_set=set()
print(f"test -- {model._non_persistent_buffers_set=}")
```
